### PR TITLE
HHH-18116 Support slow query logging for stored procedures

### PR DESCRIPTION
**Pull request by marschall**
https://hibernate.atlassian.net/browse/HHH-18116

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java [hunk 1]
```diff
@@ -652,10 +652,11 @@ private ProcedureOutputsImpl buildOutputs() {
 		}
 
 		LOG.debugf( "Preparing procedure call : %s", call);
+		String sqlString = call.getSqlString();
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java:655]:
```suggestion
		final String sqlString = call.getSqlString();
```

File: hibernate-core/src/main/java/org/hibernate/result/internal/OutputsImpl.java [hunk 2]
```diff
@@ -51,27 +53,36 @@ public class OutputsImpl implements Outputs {
...
 	}
 
 	protected ResultContext getResultContext(){
 		return context;
 	}
 
-	protected void executeStatement() {
+	protected void executeStatement(String sql) {
+		long executeStartNanos = 0;
+		if ( this.sqlStatementLogger.getLogSlowQuery() > 0 ) {
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/result/internal/OutputsImpl.java:73]:
```suggestion
		if ( sqlStatementLogger.getLogSlowQuery() > 0 ) {
```

**Review by gavinking — CHANGES_REQUESTED** [review]
Unless I'm missing something, it seems to me that it be better to pass the SQL string to the constructor of `ProcedureOutputsImpl` and then to the constructor of `OutputsImpl`, instead of passing it as an argument to the `executeStatement()` method.

**Comment by marschall** [comment]
> Unless I'm missing something, it seems to me that it be better to pass the SQL string to the constructor of `ProcedureOutputsImpl` and then to the constructor of `OutputsImpl`, instead of passing it as an argument to the `executeStatement()` method.

Fair point, corrected.

**Review by gavinking — APPROVED** [review]
LGTM, thanks!

**Pull request merged**